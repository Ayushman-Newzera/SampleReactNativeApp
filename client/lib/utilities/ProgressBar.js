var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _react=_interopRequireDefault(require("react"));var _reactNative=require("react-native");var _this=this,_jsxFileName="/Users/ayushman/Desktop/SampleReactNativeApp1/client/src/utilities/ProgressBar.js";var ProgressBar=function ProgressBar(_ref){var progress=_ref.progress;return _react.default.createElement(_reactNative.View,{style:styles.progressContainer,__self:_this,__source:{fileName:_jsxFileName,lineNumber:8,columnNumber:9}},_react.default.createElement(_reactNative.View,{style:[styles.progressBar,{width:progress*100+"%"}],__self:_this,__source:{fileName:_jsxFileName,lineNumber:9,columnNumber:13}}));};var styles=_reactNative.StyleSheet.create({progressContainer:{height:11,width:'80%',backgroundColor:'rgba(0, 0, 0, 0)',borderColor:'#000',borderWidth:2,borderRadius:5,marginTop:50},progressBar:{height:9,backgroundColor:'white',borderWidth:1.5,borderRadius:4,marginTop:-1}});var _default=ProgressBar;exports.default=_default;