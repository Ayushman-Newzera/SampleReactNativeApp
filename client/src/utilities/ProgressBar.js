// @flow

import React from 'react';
import {StyleSheet, View} from 'react-native';

const ProgressBar = ({progress}) => {
  return (
    <View style={styles.progressContainer}>
      <View style={[styles.progressBar, {width: `${progress * 100}%`}]} />
    </View>
  );
};

const styles = StyleSheet.create({
  progressContainer: {
    height: 11,
    width: '80%',
    backgroundColor: 'rgba(0, 0, 0, 0)',
    borderColor: '#000',
    borderWidth: 2,
    borderRadius: 5,
    marginTop: 50,
  },
  progressBar: {
    height: 9,
    backgroundColor: 'white',
    borderWidth: 1.5,
    borderRadius: 4,
    marginTop: -1,
  },
});

export default ProgressBar;
