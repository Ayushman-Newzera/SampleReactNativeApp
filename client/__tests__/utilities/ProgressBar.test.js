import {shallow} from 'enzyme';
import React from 'react';
import ProgressBar from '../../src/utilities/ProgressBar';

describe('ProgressBar', () => {
  it('ProgressBar should render correctly', () => {
    const component = shallow(<ProgressBar />);
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
