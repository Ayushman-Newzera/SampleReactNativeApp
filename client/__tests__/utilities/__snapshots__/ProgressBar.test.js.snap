// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar ProgressBar should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0)",
      "borderColor": "#000",
      "borderRadius": 5,
      "borderWidth": 2,
      "height": 11,
      "marginTop": 50,
      "width": "80%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 4,
          "borderWidth": 1.5,
          "height": 9,
          "marginTop": -1,
        },
        Object {
          "width": "NaN%",
        },
      ]
    }
  />
</View>
`;
